### ISSUE TITLE
Reusing prepared statements throws runtime errors

### ISSUE THREAD

**Issue by grzm** [vertica/vertica-sql-go#17]
When using prepared statements for queries, subsequent execution of the statement throws runtime errors.

```go
	var res bool

	stmt, err := connDB.PrepareContext(ctx, "SELECT true AS res")
	if err != nil {
		os.Exit(1)
	}

        // first call
	rows, err := stmt.QueryContext(ctx)
	if err != nil {
		os.Exit(1)
	}

	for rows.Next() {
		testLogger.Info("first use of stmt: %t", res) // succeeds
	}

        // second call
	rows, err = stmt.QueryContext(ctx)
	if err != nil {
		os.Exit(1)
	}

	for rows.Next() {    // <- throws
		testLogger.Info("second use of stmt: %t", res)
	}
```
For comparison, this code works with PostgreSQL (using the [`lib/pq`](https://github.com/lib/pq) driver), and this pattern of reusing a prepared statement for a query is used in the Prometheus generic database [`free/sql_exporter`](https://github.com/free/sql_exporter/blob/1a12c9586e22d3d5a8c259f47805afffbdab04ed/query.go#L109-L168) exporter.

It looks like the row description might be getting lost on subsequent calls, as the number of columns returned in the second call (from `rows.Columns()`) is 0.

I've added tests which exhibit this behavior.

https://github.com/grzm/vertica-sql-go/commit/0cdacf8e20924f854f1f012c14a23fe79c0f8d71

**Comment by huebnerr** [comment]
Thank you for finding this, @grzm. We'll work on repro'ing the error, incorporating your test into the suite, and keep you updated. Hopefully this can be fixed in short order.

**Comment by huebnerr** [comment]
@gzrm I've made a fix to the code that passes this test. Please verify that I've correctly translated your test into a new unit test. Note that in yours, the value was never Scan()'d, but I've made a change in this one to Scan and validate the cursor state.

```Go
func TestStmtReuseBug(t *testing.T) {
	connDB := openConnection(t)
	defer closeConnection(t, connDB)

	var res bool

	stmt, err := connDB.PrepareContext(ctx, "SELECT true AS res")
	assertNoErr(t, err)

	// first call
	rows, err := stmt.QueryContext(ctx)
	assertNoErr(t, err)

	defer rows.Close()

	assertNext(t, rows)
	assertNoErr(t, rows.Scan(&res))
	assertEqual(t, res, true)
	assertNoNext(t, rows)

	// second call
	rows, err = stmt.QueryContext(ctx)
	assertNoErr(t, err)

	defer rows.Close()

	assertNext(t, rows)
	assertNoErr(t, rows.Scan(&res))
	assertEqual(t, res, true)
	assertNoNext(t, rows)
}
`

**Comment by grzm** [comment]
@huebnerr Cheers! I applied your patch https://github.com/huebnerr/vertica-sql-go/commit/399816b3f198fb58f26dbab9cd135c59b0c7a72f and it resolved this particular issue.

> Note that in yours, the value was never Scan()

Thanks for catching that. Copy/paste error. That snippet was just to demonstrate in the comment. I did have full tests in the patch I linked to, where I did indeed call `rows.Scan` ;)

Thanks for the quick turnaround!
